docs/source/components.rst

```diff
@@ -0,0 +1,332 @@
+==========
+Components
+==========
+
+This document describes all components or "steps" the new Django serializers
+will have. They are closely related to existing concepts of ``django.forms`` and
+``django.core.serializers`` and borrows ideas from Django Rest Framework (DRF).
+
+Each section explains its scope and what exactly it should and shouldn't do and
+know about. It also tries to include a note about the specific step is
+currently realized in Django or DRF and gives hints how that step could be
+implemented in the future.
+
+The components are supposed to be independent from each other and it should be
+easy to customize one of it without having to touch or even braking any other
+component. Ideally you should also be able to replace a component entirely, e.g.
+with an external library.
+
+Additionally introspection of the components should be possible even within
+components. This ensures that components can be independent from each other
+while still being generic enough to make reuse of code easier.
+
+
+Construction
+============
+
+The construction step determines which fields will be used in the following
+steps. It always precedes the other steps and can't be skipped. Also once this
+step is finished the list of fields that is available to the other steps is
+frozen and *can't be changed* anymore. This means that if you need a different
+set of forms you will have to run this step again. Still, it should be easy to
+modify the set of fields on a per-request basis.
+
+Validation of any kind, loading and storing data is *not* part of this step, it
+merely constructs a data structure to be used by any of the other steps.
+
+How it is currently done
+------------------------
+
+The list of fields in a form can be specified declaratively:
+
+.. code-block:: python
+
+    class BlogPostForm(Form):
+        title = CharField(max_length=200)
+        content = CharField()
+        publish_date = DateTimeField()
+
+
+In the background this currently retains the order of the fields however that
+should not be necessary in the construction phase.
```

Comment:
I'm not sure this is something we necessarily want to lose. More relevant to mention is that this is done via a Metaclass which is opaque.